golem-worker-service/src/api/types.rs

```diff
@@ -0,0 +1,190 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct OpenAPISpec {
```

Comment:
Why not use one of the Rust crates that already models OpenAPI v3?

For example:

 - <https://github.com/kurtbuilds/openapiv3>
 - <https://github.com/glademiller/openapiv3>

(I am not endorsing these libraries.)

Reply:
```
// Custom OpenAPI types
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct OpenAPISpec {
    pub openapi: String,
    pub info: Info,
    pub paths: HashMap<String, PathItem>,
    pub components: Option<Components>,
    pub security: Option<Vec<SecurityRequirement>>,
}
```

As you can see, we can achieve a custom implementation by going with this instead. How about just adding them together for validation against openapiv3 crate's model?

Reply:
After careful consideration, I am gonna go with your suggestion on this too.